src/Migration/Validation/SwagMigrationValidationService.php

```diff
@@ -135,6 +138,10 @@ private function validateFields(SwagMigrationValidationContext $context): void
             throw MigrationException::unexpectedNullValue('id');
         }
 
+        if (!Uuid::isValid($context->getConvertedData()['id'])) {
+            throw MigrationException::invalidId($context->getConvertedData()['id'], $context->getEntityDefinition()->getEntityName());
```

Comment:
Good catch, this is an important check, for when we return data from a converter that doesn't contain a primary `id`, which is an implementation error by us.

But thinking about how would you troubleshoot that error given the information? Because `id` is likely `null` and you only have the `EntityName`. I think in this case you really need a way to figure out what data was passed to the converter (e.g. the source data). Would be nice if you could pass that somehow to this validation service as well and include it with the logs 🤔

Reply:
This error is catches and added to the log 

custom/plugins/SwagMigrationAssistant/src/Migration/Service/MigrationDataConverter.php:117

Reply:
Thats what i mean. Followed the wrong path.

Reply:
Converted data is already added